Assets/Scripts/Gamestate/Round.cs

```diff
@@ -0,0 +1,61 @@
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using UnityEngine;
+
+public class Round
+{
+    public PlayerStateController winner;
+    public Dictionary<PlayerStateController, List<PlayerStateController>> kills = new Dictionary<PlayerStateController, List<PlayerStateController>>();
+    public List<PlayerStateController> players = new List<PlayerStateController>();
+    public List<PlayerStateController> livingPlayers = new List<PlayerStateController>();
```

Comment:
public access 🤔🤔 - a recipe for potential future bugs!


In this case I'm all for privatisation!

Some of the fields may yet need read access externally, so maybe we can add some properties?
Getter property for `winner`, and maybe `IReadOnly...`-properties wrapping the dictionaries and lists for external access.
Otherwise, I think these in general should only be modifiable internally (i.e. private access), and through member method calls to the round itself.

Reply:
I agree that these fields might be better off private, and your suggestions seem well-thought-out.
If I remember correctly, this class was intended to be a simple storage class, @Fueredoriku had some thoughts on its responsibilities that he might want to mention here. Implementing your suggestions should make the code using this class a little cleaner in any case.

Reply:
IReadOnly can be a nice solution.

I do not want to make all these fields private as there are potentially classes 1 or more steps away from MatchManager that can be interested in things like round statistics (kills, winners, etc.), which would make acessing these fields a pain.

During a round we would need to store a private "writeable" collection (as it keeps track of the round dynamically), and then save this as public ReadOnly collection after the round has concluded and becomes a "storage only"-object.
It would require some refactoring (and a bit more code) in the Round class, but its negligble enough that we can do it to prevent abuse of the collection fields I guess 😅. (I'm atleast a fan of restricting architecture if you want to underline that something might be weird with your approach if you need unreachable access)